File: Data Structures & Algorithms/duplicate-integer/submission-0.java


class Solution {
    public boolean hasDuplicate(int[] nums) {
        Arrays.sort(nums);
        int l = 0;
        for (int r = 1; r < nums.length; r++){
            if (nums[l] != nums[r]){
                l++;
                continue;
            } else {
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        // the easiest way is to count all elements in a counter
        std::unordered_map<int, int> hashMap = {};
        for (int i = 0; i < nums.size() ; i++){
            hashMap [nums[i]] += 1;    
        }
        // then sort the map by values
        std::vector<std::pair<int, int>> v(hashMap.begin(), hashMap.end());
        std::sort(v.begin(), v.end(), [](const auto &a, const auto& b) {
            return a.second > b.second;
        });
        // return top k
        std::vector<int> topK {};
        for (int i = 0; i < k; i++) {
            topK.push_back(v[i].first);
        }
        return topK;
    }
};
